Formula/openshift-cli.rb

```diff
@@ -24,31 +23,54 @@ class OpenshiftCli < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "55eb4b18cc17eead8458ec3492dc7c4e788f6346f786220ea0aa48f47b194cae"
   end
 
-  depends_on "coreutils" => :build
   depends_on "go" => :build
-  depends_on "socat"
-
   uses_from_macos "krb5"
 
+  on_macos do
+    depends_on "heimdal" => :build
+  end
+
   def install
-    arch = Hardware::CPU.intel? ? "amd64" : Hardware::CPU.arch.to_s
-    os = OS.kernel_name.downcase
+    if build.stable?
+      ENV["OS_GIT_VERSION"] = version.to_s
+      ENV["SOURCE_GIT_COMMIT"] = Pathname.pwd.basename.to_s.delete_prefix("oc-")
+    end
 
-    # See https://github.com/golang/go/issues/26487
+    # See https://github.com/Homebrew/brew/issues/14763
     ENV.O0 if OS.linux?
 
-    system "make", "cross-build-#{os}-#{arch}", "OS_GIT_VERSION=#{version}", "SHELL=/bin/bash"
-    bin.install "_output/bin/#{os}_#{arch}/oc"
-
-    bash_completion.install "contrib/completions/bash/oc"
-    zsh_completion.install "contrib/completions/zsh/oc" => "_oc"
+    system "make", "oc", "SHELL=/bin/bash"
+    bin.install "oc"
+    generate_completions_from_executable(bin/"oc", "completion", base_name: "oc")
   end
 
   test do
+    # Grab version details from built client
+    version_raw = shell_output("#{bin}/oc version --client --output=json")
+    version_json = JSON.parse(version_raw)
+
+    # Ensure that we had a clean build tree
+    assert_equal "clean", version_json["clientVersion"]["gitTreeState"]
+
+    if build.stable?
```

Comment:
```suggestion
    if stable?
```
There is no `build` during the `test`.

Reply:
Thanks! Updated.